subworkflows/local/function.nf

```diff
@@ -0,0 +1,51 @@
+/*
+    RUN FUNCTIONAL CLASSIFICATION
+*/
+
+include { INTERPROSCAN_DATABASE } from '../../modules/local/interproscan_download'
+include { INTERPROSCAN          } from '../../modules/nf-core/interproscan/main'
+
+workflow FUNCTION {
+    take:
+    faas // tuple val(meta), path(PROKKA/PRODIGAL.out.faa)
+
+    main:
+    ch_versions                  = Channel.empty()
+    ch_interproscan_tsv          = Channel.empty()
+    ch_interproscan_db           = Channel.empty()
+    ch_interproscan_tsv_modified = Channel.empty()
+
+    ch_faa_for_interproscan = faas
+
+    if ( params.function_interproscan_db != null ) {
+        ch_interproscan_db = Channel
+            .fromPath( params.function_interproscan_db )
+            .first()
+    } else {
+        INTERPROSCAN_DATABASE ( params.function_interproscan_db_url )
+        ch_versions  = ch_versions.mix( INTERPROSCAN_DATABASE.out.versions )
+        ch_interproscan_db = ( INTERPROSCAN_DATABASE.out.db )
+    }
+
+    INTERPROSCAN( ch_faa_for_interproscan, ch_interproscan_db )
+    ch_interproscan_tsv = ch_interproscan_tsv.mix(INTERPROSCAN.out.tsv)
+
+    // Current INTERPROSCAN version 5.59_91.0 only includes 13 columns and not 15 which ampcombi expects, so we added them here
```

Comment:
What happens if INTERPROSCAN output changes :eyes: this scares me slightly to hardcode this but I'm not familiar with either tool

Reply:
If its updated the following chunk for column manipulation should be removed. Hence why i would keep that comment here for future bug fixing :)